File: Semana 6 - Python/pset6/readability.py
import cs50
import sys
import re

#Fuction for count words in text
def countWords(text):
    pattern = r"\b[a-zA-Z]+(?:'[a-zA-Z]+)?\b"
    words = re.findall(pattern,text)
    return len(words)

#Function for count letters in text
def countLetters(text):
    count = 0
    for i in text:
        if i.isalpha():
            count += 1
    return count

#Function for count phrases in text
def countPhrases(text):

    return len(re.findall(r"[.!?]+", text))

#Calculate the grade and print it
def checkgrade(text):
    words = countWords(text)
    letters = countLetters(text)
    phrases = countPhrases(text)

    L = (letters / words) * 100
    S = (phrases / words) * 100

    grade = (0.0588 * L) - (0.296 * S) - 15.8
    grade = round(grade)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")



def main():
    text = cs50.get_string("Text: ")
    checkgrade(text)




main()

File: Semana 2 - Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int compute_letters(string s);
int compute_words(string s);
int compute_sentences(string s);

int main(void){

    string s = get_string("Text: ");
    int letters = compute_letters(s);
    int words = compute_words(s);
    int sentences = compute_sentences(s);
    float L = letters /  (float) words * 100;

    float S = (sentences / (float) words) * 100;

    float indice = 0.0588 * L - 0.296 * S - 15.8;
    int grade = (int) round(indice);

    if(grade < 1){
        printf("Before Grade 1\n");
    }else if (grade > 16){
        printf("Grade 16+\n");
    }else{
        printf("Grade %i\n", grade);
    }




}

// Count the sentences
int compute_sentences(string s){
    int n = strlen(s);
    int sum = 0;
    for(int i = 0; i < n; i++){
        if(s[i] == '.' || s[i] == '!' || s[i] == '?'){
            sum += 1;
        }
    }
    return sum;
}

// Count the letters
int compute_letters(string s){
    int n = strlen(s);
    int sum = 0;
    for(int i = 0; i < n; i++){
        if(isalpha(s[i])){
            sum += 1;
        }
    }
    return sum;
}

// Count the words
int compute_words(string s){
    int n = strlen(s);
    int sum = 0;
    for(int i = 0; i < n; i++){
        if(isalpha(s[i-1]) && isspace(s[i])){
            sum += 1;
        }else if(isalpha(s[i-1]) && s[i] == '.'){
            sum += 1;
        }else if(isalpha(s[i-1]) && s[i] == ','){
            sum += 1;
        }else if(isalpha(s[i-1]) && s[i] == '!'){
            sum += 1;
        }else if(isalpha(s[i-1]) && s[i] == '?'){
            sum += 1;
        }
    }
    return sum;
}
